# Online/Offline
mapping_online: true
real_time: false

# Topics
points_topic: "/velodyne_points"
imus_topic: "/vectornav/IMU"

# Publishers
high_quality_publish: true   # true: Publishes the map without downsampling, can be slower. false: Publishes the downsampled map.  

# Extrinsics
estimate_extrinsics: true
print_extrinsics: false
initial_gravity: [0.0, 0.0, -9.807]
I_Translation_L: [1.25, 0.0, 0.0]
I_Rotation_L: [
     1.0,  0.0,  0.0,
     0.0, -1.0,  0.0,
     0.0,  0.0, -1.0
]

# Delays
empty_lidar_time: 0.1      # Should be at least [FULL_ROTATION_TIME]
real_time_delay: 0.1       # Should be at least [FULL_ROTATION_TIME] (without a modificated LiDAR driver)

# LiDAR
LiDAR_type: velodyne
stamp_beginning: false     # (As usual, 'false') Is the pointcloud's stamp the last point's timestamp (end of rotation) or the first's (beggining of rotation)?
offset_beginning: true     # (Xaloc's value, having a modified Velodyne LiDAR driver, is 'true') Is the offset with respect the beginning of the rotation (i.e. point.time ~ [0, 0.1]) or with respect the end (i.e. point.time ~ [-0.1, 0])? For more information see Issue #14: https://github.com/Huguet57/LIMO-Velo/issues/14
LiDAR_noise: 0.001
full_rotation_time: 0.1
min_dist: 4
downsample_rate: 4         # Downsampling rate: results show that this one can be up to 32 and still work, try it if you need a speedup
downsample_prec: 0.5       # Downsampling precision: Indoors, lower values (~0.2) work better. Outdoors, higher values (~0.5) lead to less degeneracy.

# IMU
imu_rate: 400
covariance_gyroscope: 6.01e-4
covariance_acceleration: 1.53e-2
covariance_bias_gyroscope: 1.54e-5
covariance_bias_acceleration: 3.38e-4

# Localizator
MAX_NUM_ITERS: 3
# LIMITS: [0.001] * 23
NUM_MATCH_POINTS: 5
MAX_DIST_PLANE: 2.0
PLANES_THRESHOLD: 5.e-2
degeneracy_threshold: 5.
print_degeneracy_values: false

# Delta refinement
Initialization:
#     # No initialization
#     times: []
#     deltas: [0.1]
    
    # With initialization
    times: [1.5, 2.0, 3.25]
    deltas: [0.1, 0.05, 0.02]

# Load/Save map
loadsave:
     map_name: "forum"   # "montmelo-IN2" / "forum"
     maps_path: "/media/andreu/Movies/ws_limovelo/maps"
     action: none   # OPTIONS: save / load / none
     # ini_pos: [-0.581022, -32.419407, -0.023406]
     ini_pos: [0., 0., 0.]